lib/usd/ui/views/TreeItem.cpp

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2019 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "TreeItem.h"
+#include "ItemDelegate.h"
+
+#include <cassert>
+
+MAYAUSD_NS_DEF {
+
+TreeItem::TreeItem(const UsdPrim& prim, Type t) noexcept
+	: ParentClass()
+	, fPrim(prim)
+	, fType(t)
+	, fCheckState(Qt::Unchecked)
+	, fVariantSelectionModified(false)
+{
+	initializeItem();
+}
+
+UsdPrim TreeItem::prim() const
+{
+	return fPrim;
+}
+
+int TreeItem::type() const
+{
+	return ParentClass::ItemType::UserType;
+}
+
+void TreeItem::setCheckState(Qt::CheckState st)
+{
+	assert(fType == Type::kLoad);
```

Comment:
More asserts. I think we probably want to avoid those in general. `TF_VERIFY()` might be a better option here too.